dotnet/src/Connectors/Connectors.Anthropic/Core/Models/AnthropicRequest.cs

```diff
@@ -141,58 +138,42 @@ private static AnthropicRequest CreateRequest(ChatHistory chatHistory, Anthropic
-                    new AnthropicToolCallContent
-                    {
-                        ToolId = toolCall.ToolUseId,
-                        FunctionName = toolCall.FullyQualifiedName,
-                        Arguments = JsonSerializer.SerializeToNode(toolCall.Arguments),
-                    }));
-                break;
-            default:
-                messages.AddRange(content.Items.Select(GetClaudeMessageFromKernelContent));
-                break;
-        }
-
-        if (messages.Count == 0)
-        {
-            messages.Add(new AnthropicTextContent(content.Content ?? string.Empty));
-        }
-
-        return messages;
+        return content.Items.Select(GetClaudeMessageFromKernelContent).ToList();
     }
 
     private static AnthropicContent GetClaudeMessageFromKernelContent(KernelContent content) => content switch
     {
-        TextContent textContent => new AnthropicTextContent(textContent.Text ?? string.Empty),
-        ImageContent imageContent => new AnthropicImageContent(
-            type: "base64",
-            mediaType: imageContent.MimeType ?? throw new InvalidOperationException("Image content must have a MIME type."),
-            data: imageContent.Data.HasValue
-                ? Convert.ToBase64String(imageContent.Data.Value.ToArray())
-                : throw new InvalidOperationException("Image content must have a data.")
-        ),
+        TextContent textContent => new AnthropicTextContent { Text = textContent.Text ?? string.Empty },
+        ImageContent imageContent => CreateAnthropicImageContent(imageContent),
         _ => throw new NotSupportedException($"Content type '{content.GetType().Name}' is not supported.")
     };
 
+    private static AnthropicImageContent CreateAnthropicImageContent(ImageContent imageContent)
+    {
+        var dataUri = DataUriParser.Parse(imageContent.DataUri);
+        if (dataUri.DataFormat?.Equals("base64", StringComparison.OrdinalIgnoreCase) != true)
+        {
+            throw new InvalidOperationException("Image content must be base64 encoded.");
+        }
+
+        return new AnthropicImageContent
+        {
+            Source = new()
+            {
+                Type = dataUri.DataFormat,
+                MediaType = imageContent.MimeType ?? throw new InvalidOperationException("Image content must have a MIME type."),
+                Data = dataUri.Data ?? throw new InvalidOperationException("Image content must have a data.")
+            }
+        };
+    }
+
     internal sealed class Message
     {
         [JsonConverter(typeof(AuthorRoleConverter))]
         [JsonPropertyName("role")]
-        public AuthorRole Role { get; set; }
+        public AuthorRole Role { get; init; }
 
         [JsonPropertyName("content")]
-        public IList<AnthropicContent> Contents { get; set; } = null!;
+        public IList<AnthropicContent> Contents { get; init; } = null!;
```

Comment:
I believe you don't need the `null!` in here after changing it to init

```suggestion
        public IList<AnthropicContent> Contents { get; init; } 
```

Reply:
Init does nothing here  to fix nullable warning. To Remove nullable I can use null!, #pragma or required (not available .net standard 2.0)

Reply:
If you accept null as the init value, this might be a nullable property then.